### DISCUSSION TITLE
Interaction with filesystem

### DISCUSSION THREAD

**Discussion by teddy-mindcompass** [101arrowz/fflate#151] [Q&A]
Hello everyone,

I want to know if it's possible for fflate to interact with the file system api ? The goal is too zip files and directories directly from the file system.

https://developer.mozilla.org/en-US/docs/Web/API/FileSystem 

https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryEntry/getFile

Thanks

**Comment by 101arrowz** [comment]
Sure, the Filesystem API gives you [`FileSystemFileEntry`](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemFileEntry)s for each file you access. On each of those, you can call `file()` to get a `File` object, which can then be converted to a `Uint8Array` or a `ReadableStream` for use with `fflate`.

There's a tutorial on building a simple ZIP creator [in the wiki](https://github.com/101arrowz/fflate/wiki/Guide:-Modern-(Buildless)), you should be able to adapt it to work with the Filesystem API pretty easily, but let me know if you need more help.
